src/Billing/X12ClaimRepost.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/**
+ *  package OpenEMR
+ *  link    https://www.open-emr.org
+ *  author  Sherwin Gaddis <sherwingaddis@gmail.com>
+ *  Copyright (c) 2022.
+ *  license https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Billing;
+
+use phpseclib\Net\SFTP;
+
+class X12ClaimRepost
+{
+    private bool $status;
+    public function __construct(
+        $host,
+        $username,
+        $password,
+        $port = 22
+    ) {
+        $connection = new SFTP($host, $port);
+        $this->status = $connection->login($username, $password);
+    }
+    //ToDo: fix this for more than one SFTP account. Right now, it will only select the first one in the table.
+    public function getConnectionStatus()
+    {
+        return $this->status;
+    }
+    public static function x12Url()
+    {
+        return sqlQuery('SELECT x12_sftp_host FROM x12_partners');
+    }
+
+    public static function x12Username()
+    {
+        return sqlQuery('SELECT x12_sftp_login FROM x12_partners');
+    }
+    public static function x12Password()
+    {
+        return sqlQuery('SELECT x12_sftp_pass FROM x12_partners');
+    }
+
+    public static function updateStatus(): void
+    {
+        sqlQuery("UPDATE `x12_remote_tracker` SET status = 'waiting', messages = NULL WHERE `status` = 'login-error'");
```

Comment:
hi @juggernautsei, this might be too broad of a query. If there was a way to give a button some capability on each line of the billing tracker to call this repost then it wouldn't risk resending a file with a batch of claims that may have been regenerated from the billing manager and successfully sent in place of the file that failed.

Reply:
I understand what you are asking. The thought process behind the build was that this would run automatically and not be a manual process. That is why I put the file in the library to run it in the background manager. I added it to the claim tracker page so that when the claim tracker page is looked at and the connection is restored. The query would reset all the claims that have been marked with the login error (which was our case). The users were not regenerating any of the batches. In fact they did not want to go through all the hassle of regenerating batches because of the failed login. They just wanted the batches to go out once the connection has been restored with no effort on their part. That was the driving force behind this feature.

If a batch is resent, the clearinghouse will just reject the claims. The claim won't be processed again to my knowledge. We have done that. Also, the billing manager does not auto batch so why would there be a duplicate batch that has successfully gone through while the SFTP connect is down?

If someone while the SFTP connection is down manually uploaded a file the clearinghouse and the batch failed. When the connection returns the batch will be the same one that was uploaded manually during the time that the connection was down.  

So, compromise. Make it a global to resend all claims that are marked login-error.

Reply:
hi @juggernautsei, since it should be fixed for more than 1 sftp account then having a button on each line would be a good solution

the button could also help fix other errors like bad parameters etc